### ISSUE TITLE
Static analysis by Clang shows memory use after it has been freed.

### ISSUE THREAD

**Issue by Tiza59** [macroxue/bridge-solver#3]
T& operator[](size_t i) { return reinterpret_cast<T*>(items)[i]; }

Sorry to report that running the DDS with -r crashes with seg fault, maybe that is the reason for the issue above?

**Comment by macroxue** [comment]
Thanks for reporting. I was able to reproduce the crash with Clang MemorySanitizer `-fsanitize=memory`. I will take a deeper look.
```
==289049==WARNING: MemorySanitizer: use-of-uninitialized-value                                                                                                                
    #0 0x55627a4d475c in Pattern::Append(Vector<Pattern>&)
    #1 0x55627a4d3e73 in Pattern::UpdateBounds(Bounds)
    #2 0x55627a4cbaf1 in Pattern::Update(Pattern&)
    ...
```

**Comment by macroxue** [comment]
Fixed by 26e3aece2463733a2617556b133ed6ff8701985f

**Comment by Tiza59** [comment]
Thank you for fixing this issue. Doesn't crash anymore on my Mac.
